src/RemoteSystemdFileManager.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\Deployer\Systemd;
+
+use function Deployer\run;
+use function Safe\preg_split;
+use function Safe\sprintf;
+use Webmozart\Assert\Assert;
+
+final class RemoteSystemdFileManager
+{
+    private static string $defaultPath = '{{systemd_local_path}}';
+
+    /**
+     * Returns a string like my-service---prod---r34.service
+     */
+    public static function getRemoteFilename(string $filename, string $stage, string $release): string
+    {
+        $pathInfo = pathinfo($filename);
+        Assert::keyExists($pathInfo, 'extension');
+        Assert::stringNotEmpty($pathInfo['extension']);
+
+        return sprintf('%s---%s---r%s.%s', $pathInfo['filename'], $stage, $release, $pathInfo['extension']);
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getAll(string $path = null): array
+    {
+        $path = $path ?? self::$defaultPath;
+
+        $output = run(sprintf('find %s -type f -name "*.service"', $path));
+
+        return self::handleOutput($output);
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getByStage(string $stage, string $path = null): array
+    {
+        $path = $path ?? self::$defaultPath;
+
+        $output = run(sprintf('find %s -type f -name "*---%s---*.service"', $path, $stage));
```

Comment:
I would suggest to define some const like `FILE_TEMPLATE = '%s---%s---%s.service';` and use it with '*' or actual value at `sprintf`.

Reply:
The file template is not the same on every case. I just updated the PR with tests. Can you skim through those?